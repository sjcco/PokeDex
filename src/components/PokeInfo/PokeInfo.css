.container {
  display: flex;
  height: auto;
  justify-content: center;
  border-radius: 0.5rem;
  box-shadow: 0.05rem 0.1rem 0.3rem -0.03rem rgba(0, 0, 0, 0.45);
  margin: 0.5rem 0.5rem;
  background-color: white;
}

.info {
  width: 75%;
  display: flex;
  flex-direction: column;
}

.info-header {
  width: 100%;
  display: flex;
  justify-content: center;
}

.info-header h1 {
  width: 50%;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-size: 1.3rem;
  line-height: 28px;
  text-align: center;
}

.info-header p {
  width: 40%;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-size: 1.3rem;
  font-weight: 100;
  line-height: 28px;
  text-align: center;
}

.card-icon {
  width: 25%;
  height: auto;
  margin: auto 0.5rem auto 0;
}

.badges-container {
  width: auto;
  margin: 0 auto 0.5rem auto;
}

.genus-title {
  margin: 0 auto 1rem auto;
  width: auto;
  font-size: 1rem;
  font-weight: 200;
  font-family: 'Trebuchet MS', Arial, sans-serif;
}
