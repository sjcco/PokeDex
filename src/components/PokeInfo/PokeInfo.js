import React from 'react';
import PropTypes from 'prop-types';
import {
  container, info, infoHeader, cardIcon, badgesContainer, genusTitle,
} from './PokeInfo.css';
import TypeBadge from '../TypeBadge/TypeBadge';
import { padLeadingZeros } from '../../helpers';

const PokeInfo = ({
  icon, pokeName, id, genus, type1, type2,
}) => (
  <div className={container}>
    <div className={info}>
      <div className={infoHeader}>
        <h1>{pokeName}</h1>
        <p>{`#${padLeadingZeros(id, 4)}`}</p>
      </div>
      <h2 className={genusTitle}>{genus}</h2>
      <div className={badgesContainer}>
        <TypeBadge type={type1} />
        {!(type2 === 'null') && <TypeBadge type={type2} />}
      </div>
    </div>
    <img className={cardIcon} src={icon} alt="" />
  </div>
);

PokeInfo.propTypes = {
  icon: PropTypes.string.isRequired,
  pokeName: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  genus: PropTypes.string.isRequired,
  type1: PropTypes.string.isRequired,
  type2: PropTypes.string.isRequired,
};

export default PokeInfo;
