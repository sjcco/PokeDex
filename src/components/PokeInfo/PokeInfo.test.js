/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import PokeInfo from './PokeInfo';

test('renders content', () => {
  const component = renderer.create(<PokeInfo
    icon="icon"
    pokeName="name"
    id="id"
    genus="genus"
    type1="type1"
    type2="type2"
  />).toJSON();
  expect(component).toMatchSnapshot();
});
