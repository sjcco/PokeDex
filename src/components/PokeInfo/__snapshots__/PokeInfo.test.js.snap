// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
<div
  className="container"
>
  <div
    className="info"
  >
    <div
      className="infoHeader"
    >
      <h1>
        name
      </h1>
      <p>
        #00id
      </p>
    </div>
    <h2
      className="genusTitle"
    >
      genus
    </h2>
    <div
      className="badgesContainer"
    >
      <div
        className="pkmnType"
        style={
          Object {
            "backgroundColor": null,
          }
        }
      >
        <span>
          type1
        </span>
      </div>
      <div
        className="pkmnType"
        style={
          Object {
            "backgroundColor": null,
          }
        }
      >
        <span>
          type2
        </span>
      </div>
    </div>
  </div>
  <img
    alt=""
    className="cardIcon"
    src="icon"
  />
</div>
`;
